modules/sc-mesh-secure-deployment/src/nats/src/cbma_adaptation.py

```diff
@@ -176,35 +244,57 @@ def __create_vlan_interfaces(self) -> None:
                         f"VLAN interface {vlan_name} created and configured."
                     )
                 except subprocess.CalledProcessError as e:
+                    success = False
                     self.logger.error(f"Error creating VLAN interface {vlan_name}: {e}")
             else:
+                success = False
                 self.logger.error(f"Invalid configuration for VLAN {vlan_name}.")
+        return success
```

Comment:
I'm defaulting success variable as True at the beginning of function and setting it False in case of any trials within for loop fails. So the function will try to create all interfaces per config thus not bailing out after first failure. Return value check is merely for unit testing purposes for now.